src/java/org/apache/cassandra/index/sai/disk/v1/keystore/KeyLookup.java

```diff
@@ -197,43 +197,34 @@ public long clusteredSeekToKey(ByteComparable key, long startingPointId, long en
             updateCurrentBlockIndex(startingPointId);
             resetToCurrentBlock();
 
-            if (compareKeys(currentKey, skipKey) == 0)
-                return startingPointId;
+            // We can return straightaway if the currentPointId is within the requested partition range and the keys match
+            if (currentPointId >= startingPointId && currentPointId < endingPointId && compareKeys(currentKey, skipKey) == 0)
+                return currentPointId;
```

Comment:
So it is possible here that after moving to `startingPointId`'s block, we might have a target/skip key that is the first key in the block, but we've restricted the point ID range to start after that key's point ID?

Reply:
Correct. It's just an optimisation on the off chance that this is the case.